### ISSUE TITLE
Unable to download (blocked?) Youku videos! Please help! Thank you!  *Edited

### ISSUE THREAD

**Issue by dcl9** [soimort/you-get#2099]
#Please make sure these boxes are checked before submitting your issue – thank you!

- [ Yes] You can actually watch the video in your browser or mobile application, but not download them with `you-get`.
- [ Yes???] Your `you-get` is up-to-date.
- [ Have trouble understanding] I have read <https://github.com/soimort/you-get/wiki/FAQ> and tried to do so.
- [I don't know ] The issue is not yet reported on <https://github.com/soimort/you-get/issues> or  <https://github.com/soimort/you-get/wiki/Known-Bugs>. If so, please add your comments under the existing issue.
- [ Yes] The issue (or question) is really about `you-get`, not about some other code or project.

Run the command with the `--debug` option, and paste the full output inside the fences:

```
Hi
I came across You-Get from watching this Youtube tutorial video:

https://www.youtube.com/watch?v=7zba-be5qiU
(The video is a bit outdated.)
#
I do not know anything about programming.
After some trial and error, I managed to follow and complete the steps in the video.
*I changed Input Variable name: Path FFmpeg and Variable value: C:\ffmpeg\bin - OK to
Input Variable name: Path FFmpeg and Variable value: C:\ffmpeg-3.3.2\bin - OK

I am unsure if I have downloaded the correct versions of Python, ffmpeg and rtmpdump so I have attached screenshots of my C drive where everything is installed and extracted.
I also attached a screenshot of the folder Python 3.6.1 
>>I am unable to attach the screenshot files...

ffmpeg-3.3.2.tar
python-3.6.1-embed-win32
python36
rtmpdump-2.3-windows
you-get-0.4.715-win32-full

I tried downloading a Youku video and got an error message.
To rule out other factors, I tried downloading a Youtube video and the attempt was successful.
Thus, I am here asking for help about downloading Youku videos.
Thank you so much in advance and please kindly advise using easier language as I have no experience with programming! Thank you twice for your patience.

Below are my logs:
It says the video is blocked but I am able to watch the video with the Unblock Youku extension?
Do I have to use a special code or prompt?
Please advise.

you-get: version 0.4.715, a tiny downloader that scrapes the web.

>chcp 65001
Active code page: 65001

>you-get -i http://v.youku.com/v_show/id_XMjgzNzQ1Nzg2NA==.html
you-get: [Failed] 客户端无权播放,201
you-get: [error] oops, something went wrong.
you-get: don't panic, c'est la vie. please try the following steps:
you-get:   (1) Rule out any network problem.
you-get:   (2) Make sure you-get is up-to-date.
you-get:   (3) Check if the issue is already known, on
you-get:         https://github.com/soimort/you-get/wiki/Known-Bugs
you-get:         https://github.com/soimort/you-get/issues
you-get:   (4) Run the command with '--debug' option,
you-get:       and report this issue with the full output.

>you-get --debug http://v.youku.com/v_show/id_XMjgzNzQ1Nzg2NA==.html
[DEBUG] get_content: http://play.youku.com/play/get.json?vid=XMjgzNzQ1Nzg2NA==&ct=10
[DEBUG] get_content: http://play.youku.com/play/get.json?vid=XMjgzNzQ1Nzg2NA==&ct=12
you-get: [Failed] 客户端无权播放,201
you-get: version 0.4.715, a tiny downloader that scrapes the web.
you-get: ['http://v.youku.com/v_show/id_XMjgzNzQ1Nzg2NA==.html']
Traceback (most recent call last):
  File "src\you_get\extractors\youku.py", line 180, in prepare
AssertionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 11, in <module>
  File "src\you_get\__main__.py", line 92, in main
  File "src\you_get\common.py", line 1429, in main
  File "src\you_get\common.py", line 1342, in script_main
  File "src\you_get\common.py", line 1158, in download_main
  File "src\you_get\common.py", line 1422, in any_download
  File "src\you_get\extractor.py", line 41, in download_by_url
  File "src\you_get\extractors\youku.py", line 200, in prepare
  File "src\you_get\util\log.py", line 98, in wtf
NameError: name 'exit' is not defined
you-get returned -1

>


```

If there's anything else you would like to say (e.g. in case your issue is not about downloading a specific video; it might as well be a general discussion or proposal for a new feature), fill in the box below; otherwise, you may want to post an emoji or meme instead:

> [WRITE SOMETHING]
> [OR HAVE SOME :icecream:!]

汉语翻译最终日期：2016年02月26日

在提交前，请确保您已经检查了以下内容!

- [ ] 你可以在浏览器或移动端中观看视频，但不能使用`you-get`下载.
- [ ] 您的`you-get`为最新版.
- [ ] 我已经阅读并按 <https://github.com/soimort/you-get/wiki/FAQ> 中的指引进行了操作.
- [ ] 您的问题没有在<https://github.com/soimort/you-get/issues> , <https://github.com/soimort/you-get/wiki/FAQ> 或 <https://github.com/soimort/you-get/wiki/Known-Bugs> 报告，否则请在原有issue下报告.
- [ ] 本问题确实关于`you-get`, 而不是其他项目.

请使用`--debug`运行，并将输出粘贴在下面:

```
[在这里粘贴完整日志]
```

如果您有其他附言，例如问题只在某个视频发生，或者是一般性讨论或者提出新功能，请在下面添加;或者您可以卖个萌:

> [您的内容]
> [舔 :icecream:!]

**Comment by rosynirvana** [comment]
Because youku changed their API.

Please delete all windows specific you-get release, F.I. the outdated ```you-get-0.4.715-win32-full``` or ```you-get``` installed with Chocolatey. Install ```you-get``` with ```pip3``` (It could be ```pip``` for some windows python platform). Then apply this #2015 .

You need a proxy for some youku videos. It's reported that ```unblock youku``` pac won't work.

**Comment by dcl9** [comment]
hi rosynirvana

Thank you so much for your reply.
I am unsure if I understood you correctly as I have no programming knowledge.
I thank you for your patience as it must be frustrating to explain some things that are so basic to you.

>>Because youku changed their API.

I don't know what API is.  Is it necessary to understand? > <

>>Please delete all windows specific you-get release, F.I. the outdated you-get-0.4.715-win32-full or you-get installed with Chocolatey. Install you-get with pip3 (It could be pip for some windows python platform). Then apply this #2015 .

I deleted the you-get I installed.  
When you say "install you-get with pip3," what do you mean?
I went to this page:  https://pypi.python.org/pypi/you-get#downloads
I downloaded:
you-get-0.4.750.tar.gz (md5, pgp) 	Source 		2017-05-21 	118KB
you_get-0.4.750-py3-none-any.whl (md5, pgp) 	Python Wheel 	3.6 	2017-05-21 	166KB
I extracted the tar.gz file.

After extracting, I don't see any bat file and exe? file as I did with the previous "you-get-0.4.715-win32-full".
How do I then go about downloading Youku videos?

>>Then apply this #2015 
I don't understand what this means...so sorry

Can you elaborate on the prompts/commands in steps for downloading a Youku video?
Please help~
Your help is greatly appreciated.

>>You need a proxy for some youku videos. It's reported that unblock youku pac won't work.

I know what a proxy is but I don't know how to apply or get one.
Or do you need to download a software/application?
Please kindly advise.

Again, I know the questions I am asking may be really silly to you...
please be patient with me?

Thanks so much!!!
=D


You need a proxy for some youku videos. It's reported that unblock youku pac won't work.

**Comment by dcl9** [comment]
These are the steps I followed in the Youtube tutorial video:

Download:
- Python: https://www.python.org/downloads/
- FFmpeg: https://www.ffmpeg.org/
- RTMPDump: http://rtmpdump.mplayerhq.hu/download/
- You-Get: https://you-get.org/
Setup:
- Download and install Python on C:\
- Download and extract FFmpeg, RTMPDump on C:\ same C:\ffmpeg, C:\rtmpdump-2.3
Set path: Right click on My Computer select Properties - Advanced system setting - Environment Variables…
New… Input Variable name: Path FFmpeg and Variable value: C:\ffmpeg\bin - OK
New… Input Variable name: Path RTMPDump and Variable value: C:\rtmpdump-2.3 - OK
OK to save.
- Download You-Get, extract it and move to Python folder


Use You-Get download Videos, Images, Audios**<<I am now stuck here since I don't see any .bat file?**
Run you-get.bat on C:\Python36-32\you-get
Input: chcp 65001 and press Enter
Input: you-get -i download link and press Enter to see all available quality and formats
Input: you-get --format=... download link and press Enter to start download.

**Comment by rosynirvana** [comment]
Please check the main page

> Option 1: Install via pip
> The official release of you-get is distributed on PyPI, and can be installed easily from a PyPI mirror via the pip package manager. Note that you must use the Python 3 version of pip:
> $ pip3 install you-get

for some python platforms on windows it could be ```pip``` instead of ```pip3```

Then apply the patch
```
pip3 install --upgrade https://github.com/rosynirvana/you-get/archive/youku.zip
```

**Comment by soimort** [comment]
Closed via #2015.
